test/unit/ptable_test.rb

```diff
@@ -101,20 +101,48 @@ def setup
       end
     end
 
-    context 'map_oses' do
+    context 'map_metadata' do
       test 'returns OSes that are in the db' do
         metadata = { 'oses' => ['centos 5.3', 'Fedora 19'] }
-        assert_equal [Operatingsystem.find_by_title('centos 5.3')], Ptable.map_oses(metadata)
+        assert_equal [Operatingsystem.find_by_title('centos 5.3')], Ptable.map_metadata(metadata, 'oses')
+      end
+
+      test 'returns Orgs that are in the db' do
+        metadata = { 'organizations' => ['Organization 1', 'Organization One'] }
+        assert_equal [Organization.find_by_name('Organization 1')],
+                     Ptable.map_metadata(metadata, 'organizations')
+        # test organisations as well as organizations metadata
+        metadata = { 'organisations' => ['Organization 1', 'Organization One'] }
+        assert_equal [Organization.find_by_name('Organization 1')],
+                     Ptable.map_metadata(metadata, 'organisations')
+      end
+
+      test 'returns Locations that are in the db' do
+        metadata = { 'locations' => ['Location 1', 'Location One'] }
+        assert_equal [Location.find_by_name('Location 1')],
+                     Ptable.map_metadata(metadata, 'locations')
       end
 
       test 'returns an empty array for no matched OSes' do
         metadata = { 'oses' => ['Fedora 19'] }
-        assert_equal [], Ptable.map_oses(metadata)
+        assert_equal [], Ptable.map_metadata(metadata, 'oses')
       end
 
-      test 'defaults to an emtpy array' do
+      test 'returns an empty array for no matched Org' do
+        metadata = { 'organizations' => ['Organization One'] }
+        assert_equal [], Ptable.map_metadata(metadata, 'organizations')
+      end
+
+      test 'returns an empty array for no matched Location' do
+        metadata = { 'locations' => ['Location One'] }
+        assert_equal [], Ptable.map_metadata(metadata, 'locations')
+      end 
+
+      test 'map_metadata defaults to an emtpy array' do
         metadata = {}
-        assert_equal [], Ptable.map_oses(metadata)
+        [ 'oses', 'locations', 'organizations' ].each do |param|
+          assert_equal [], Ptable.map_metadata(metadata, param)
+        end
       end
     end
   end
```

Comment:
I don't see any additions to the tests to show a template actually gets a taxonomy assigned?